### PR TITLE
parse traces from the common clock infrastructure

### PR DESCRIPTION
**Pull request by kdub**
Traces from clock_set_rate, clock_enable, clock_disable need
specialized parsing. Add additional feature to parse these traces.

Test: run the 3 new unit tests (nosetests tests/test_common_clk.py)

Change-Id: Ib93b72697fc4d5eb30cffb914bbe0cb4c4cd872d

### CODE REVIEW

File: trappy/common_clk.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+from trappy.base import Base
+from trappy.dynamic import register_ftrace_parser, register_dynamic_ftrace
+
+#clock traces are of the form "clk_name field0=x field1=y ..."
+def parse_common_clock_data(data_str, fn):
+    data_l = data_str.split(' ', 1)
+    p = fn.generate_data_dict(data_l[1])
+    p.update({"clk_name" : data_l[0]})
+    return p
+
```
Comment (bjackman) [inline trappy/common_clk.py]: I think you can simplify this: if you add an intermediate common class then you only need to write generate_data_dict once. And then you can actually move the `parse_common_clock_data` inside that method. Something like this (untested):

```python
class CommonClkBase(Base):
    def generate_data_dict(self, data_str):
        #clock traces are of the form "clk_name field0=x field1=y ..."
        clk_name, fields = data_str.split(' ', 1)
        ret = super(CommonClkBase, self).generate_data_dict(fields)
        ret['clk_name'] = clk_name
        return ret
```

**Comment by kdub** [comment]
@bjackman Thanks for the review, adopted all suggestions

**Comment by bjackman** [comment]
Welcome aboard :grin:

**Comment by joelagnel** [comment]
Awesome. :):)

**Pull request merged**